lib/ansible/modules/cloud/amazon/aws_organizations_facts.py

```diff
@@ -0,0 +1,151 @@
+- name: Facts are published in ansible_facts['organizations']
+  debug:
+    msg: "{{ ansible_facts['organizations'] }}"
+'''
+
+RETURN = r'''
+organizations:
+  description: AWS Organization Facts.
+  returned: success
+  type: dict
+  sample:
+    id: o-abcde01234
+    arn: arn:aws:organizations::123456789012:organization/o-abcde01234
+    feature_set: ALL
+    master_account_arn: arn:aws:organizations::123456789012:account/o-abcde01234/123456789012
+    master_account_id: 123456789012
+    master_account_email: mailbox@example.com
+    available_policy_types:
+      - Type: SERVICE_CONTROL_POLICY
+        Status: ENABLED
+    accounts:
+      returned: success
+      type: list
+      sample:
+        - arn: arn:aws:organizations::210987654321:account/o-abcde01234/210987654321
+          email: mailbox+account@example.com
+          id: '210987654321'
+          joined_method: CREATED
+          joined_timestamp: 2019-01-01 16:33:33.333000
+          name: Another AWS Account
+          status: ACTIVE
+        - arn: arn:aws:organizations::123456789012:account/o-abcde01234/123456789012
+          email: mailbox@example.com
+          id: '123456789012'
+          joined_method: CREATED
+          joined_timestamp: 2018-01-01 15:22:22.222000
+          name: Master AWS Account
+          status: ACTIVE
+'''
+from ansible.module_utils.aws.core import (AnsibleAWSModule, HAS_BOTO3)
+from ansible.module_utils.ec2 import (AWSRetry, camel_dict_to_snake_dict)
+try:
+    from botocore.exceptions import (BotoCoreError, ClientError)
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def describe_organization(client, aws_retry=True):
+    return client.describe_organization(aws_retry=aws_retry)['Organization']
+
+
+def list_accounts(client):
+    paginator = client.get_paginator('list_accounts')
+    return paginator.paginate().build_full_result()['Accounts']
+
+
+def main():
+    argument_spec = dict(all_facts=dict(type='bool',
+                                        required=False,
+                                        default=False),
+                         accounts=dict(type='bool',
+                                       required=False,
+                                       default=False))
+    module = AnsibleAWSModule(argument_spec=argument_spec,
+                              supports_check_mode=True)
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    client = module.client('organizations',
+                           retry_decorator=AWSRetry.jittered_backoff(
+                               retries=10, delay=3, max_delay=30))
+
+    result = {'ansible_facts': {}}
+
+    try:
+        organizations = describe_organization(client)
+        result['ansible_facts']['organizations'] = camel_dict_to_snake_dict(
+            organizations)
+
+        # Optional outputs
+        all_facts = module.params.get('all_facts')
+        if all_facts or module.params.get('accounts'):
```

Comment:
Why do you have two separate parameters with identical functionality?

Reply:
This looks like the logic from one of the other _info/_fact modules (I forget which).  The idea is that in the long run you add
- accounts
- policies
- (something else)

and all_facts lists them all.  The advantage here comes when you don't have full read-access, and can only read certain attributes.  I'm not sure it makes sense without also supporting listing the SCPs attached to the OU